src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -109,11 +109,13 @@ public void Dispose()
         {
             _shutdownBlock.Set();
 
-            AppDomain.CurrentDomain.ProcessExit -= OnProcessExit;
+            UnregisterShutdownHandlers();
             Console.CancelKeyPress -= OnCancelKeyPress;
```

Comment:
Should this register for the SIGINT and SIGQUIT signals directly as well, and not go through the extra console layer on .NET 6?

Reply:
My thinking was that the platform has `Console.CancelKeyPress`, we should use it. I don't think we would suggest to others to change their usage of `Console.CancelKeyPress` to use PosixSignalRegistration.

I think registering SIGTERM is a special case here.

It would also mean more of a split between .NET 6 and previous TFMs.

Reply:
Yes, it does. Thanks for the suggestion.